utils/parse_pci_dss_standard.py

```diff
@@ -0,0 +1,228 @@
+
+    # drop notes separators from from description to get title
+    def set_title(self, title_candidate):
+        if title_candidate == None:
+            return
+        for splitter in ComplianceControlNode.title_split_patterns:
+            title_match = re.match(splitter, title_candidate)
+            if title_match == None:
+                continue
+            title_candidate = title_match[1]
+            break
+        self.title = re.sub("\n", "", title_candidate.strip()) # one line title
+        self.title = re.sub(",$", ".", self.title) # should end with stop
+        self.title = re.sub("^[A-Z0-9\.]+\s","",self.title) # id already has its field
+
+    def add_subcontrol(self, node):
+        self.controls.append(node)
+
+    def get_id(self):
+        return self.id
+
+    def controls(self):
+        return self.controls
+
+    def is_child_of(self, parent_candidate_id):
+        if parent_candidate_id in self.id:
+            return True
+        return False
+
+    def choose_parent(self, parent_candidate):
+        while True:
+            if parent_candidate == None or \
+               parent_candidate.get_id() == None:
+                return self.root
+            if parent_candidate.get_id() and \
+               parent_candidate.get_id() in self.id:
+                return parent_candidate
+            parent_candidate = parent_candidate.parent
+
+    def choose_next(self, visited):
+        if len(self.controls) > 0 and not ( self.controls[0] in visited):
+            return self.controls[0]
+        current_idx = self.parent.controls.index(self)
+        if current_idx+1 < len(self.parent.controls):
+            return self.parent.controls[current_idx+1]
+        elif self.parent.get_id() == None:
+            return None
+        else:
+            return self.parent.choose_next(visited)
+
+    def wrap_line(self, line,  indent_length):
+        if not line:
+            return ""
+        extra_indent = indent_length+len("  title: ") 
+        if (len(line) + extra_indent) <= self.wrap_length:
+            return line
+        wrap_line_length = self.wrap_length - extra_indent
+        last_space = line[:wrap_line_length].rindex(' ')
+        tail = line[(last_space+1):]
+        line = line[:(last_space+1)]+"\n"+indent_length*" "+self.wrap_line(tail, indent_length)
+        return line
+
+
+    def print(self):
+        print("%s- id: %s" % (self.indent*self.depth, self.id))
+        print("%s  title: '%s'" % (self.indent*self.depth, self.wrap_line(self.title, (self.depth+2)*len(self.indent))))
+        print("%s  levels: " % (self.indent*self.depth))
+        print("%s    - base" % (self.indent*self.depth))
+        print("%s  status: not applicable" % (self.indent*self.depth))
+        if(len(self.controls) > 0):
+            print("%s  controls:" % (self.indent*self.depth))
+        else:
+            print("%s  rules: []" % (self.indent*self.depth))
+            print("")
+
+
+class ComplianceControlTree:
```

Comment:
I have a similar concept in my script.

I thought about pulling something like this into its own library. If I were to do that, would you be interested in using it?

It might look something like:

```python

b = Benchmark(name='PCI-DSS', version='4.0')
# parse sections from PDF
for section in sections:
    s = Section(id=section['id'], title=section['title'], description=section['description'])
    b.insert_section(s)

... 
# control parsing
```

Reply:
I threw what I have in a library - https://github.com/rhmdnd/pycompliance

I updated my script to use it https://github.com/ComplianceAsCode/content/pull/10455

If you want to use this and if other folks here are OK with the approach, I can propose a release to PyPI and add it to the python dependencies.

Reply:
The pycompliance library was added to the project requirements and I used it in a similar script. Feel free to leverage it here if it's useful for you, too.

https://github.com/ComplianceAsCode/content/blob/master/utils/generate_profile.py#L14
https://github.com/ComplianceAsCode/content/blob/master/requirements.txt#L12